rust/src/smb/smb2_records.rs

```diff
@@ -521,18 +522,15 @@ named!(pub parse_smb2_response_record<Smb2Record>,
            })
 ));
 
-#[derive(Debug,PartialEq)]
-pub struct SmbRecordPostGap<'a> {
-    pub data: &'a[u8],
+pub fn search_smb_record<'a>(i: &'a [u8]) -> nom::IResult<&'a [u8], &'a [u8]> {
+    let mut d = i;
+    while d.len() >= 4 {
+        if d[1] == 'S' as u8 && d[2] == 'M' as u8 && d[3] == 'B' as u8 &&
```

Comment:
Nitpicking:
- you may prefer the notation `b'S'` instead of `'S' as u8` for byte literals
- the first tests can be simplified to become `if &d[1..4] == b"SMB"`

Reply:
Will push a new version soon.